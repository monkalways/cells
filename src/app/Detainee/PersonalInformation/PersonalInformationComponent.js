import React from 'react';
import PropTypes from 'prop-types';
import {
  Grid, Paper, Typography, withStyles,
} from '@material-ui/core';
import detaineeImage from './detainee.PNG';
import PersonalStats from './PersonalStats';

const propTypes = {
  classes: PropTypes.shape({}).isRequired,
  detainee: PropTypes.shape({}).isRequired,
};

const PersonalInformationComponent = ({ classes, detainee }) => {
  console.log(detainee);

  const getDetaineeImage = () => {
    const { intakePhotoResourceUri } = detainee;
    if (intakePhotoResourceUri) {
      return intakePhotoResourceUri;
    }
    return detaineeImage;
  };

  return (
    <React.Fragment>
      <Grid
        container
        direction="column"
        justify="space-between"
        className={classes.column}
        spacing={8}
      >
        <Grid item>
          <Typography>Personal Information</Typography>
        </Grid>
        <Grid item>
          <Grid container alignItems="flex-start" justify="flex-start">
            <Grid item sm={4}>
              <img
                className={classes.img}
                alt="Detainee"
                src={getDetaineeImage()}
              />
            </Grid>
            <Grid item sm={8}>
              <PersonalStats detainee={detainee} />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
};

PersonalInformationComponent.propTypes = propTypes;

export default withStyles((theme) => ({
  img: {
    width: '150px',
    height: '150px',
  },
  row: {
    padding: theme.spacing.unit * 0.5,
  },
  column: {
    padding: theme.spacing.unit * 0.5,
  },
}))(PersonalInformationComponent);
