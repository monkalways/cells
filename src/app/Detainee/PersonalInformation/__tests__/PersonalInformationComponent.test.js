import React from 'react';
import { shallow } from 'enzyme';

import { PersonalInformationComponent } from '../PersonalInformationComponent';

describe('PersonalInformationComponent', () => {
  let detainee;

  beforeEach(() => {
    detainee = {
      detentionUnitName: 'detentionUnitName',
      dob: 'dob',
      firstName: 'firstName',
      gender: 'gender',
      lastName: 'lastName',
      propertyBagNumber: 'propertyBagNumber',
    };
  });

  const setup = () => {
    const classes = {
      img: 'img',
      test: 'text',
    };

    return shallow(<PersonalInformationComponent classes={classes} detainee={detainee} />);
  };

  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render "detainee.PNG" image if detainee photo is not available', () => {
    const wrapper = setup();

    expect(wrapper.find('img')).toHaveProp('src', 'detainee.PNG');
  });

  it('should render detainee photo if authenticated and the detainee photo is available', () => {
    detainee.intakePhotoResourceUri = 'john-smith-intake-photo.JPEG';
    const wrapper = setup();

    expect(wrapper.find('img')).toHaveProp(
      'src',
      'john-smith-intake-photo.JPEG',
    );
  });
});
