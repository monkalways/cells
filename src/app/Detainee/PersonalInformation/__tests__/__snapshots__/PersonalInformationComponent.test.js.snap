// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInformationComponent should render 1`] = `
<Fragment>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <WithStyles(Typography)
        variant="h5"
      >
        Personal Information
      </WithStyles(Typography)>
    </WithStyles(Grid)>
    <WithStyles(Grid)
      container={true}
      spacing={24}
    >
      <WithStyles(Grid)
        item={true}
        xs={3}
      >
        <img
          alt="Detainee"
          className="img"
          src="detainee.PNG"
        />
      </WithStyles(Grid)>
      <WithStyles(Grid)
        item={true}
        xs={9}
      >
        <WithStyles(PersonalStatsComponent)
          detainee={
            Object {
              "detentionUnitName": "detentionUnitName",
              "dob": "dob",
              "firstName": "firstName",
              "gender": "gender",
              "lastName": "lastName",
              "propertyBagNumber": "propertyBagNumber",
            }
          }
        />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</Fragment>
`;
